### PR TITLE
Allow a Taxon to be saved without a linked parent taxon

### PR DESCRIPTION
**Pull request by whoojemaflip**
It should be possible to save a Taxon in content-tagger without
a parent taxon. Previously, publishing-api would error on the patch-links call.

This makes it possible to save a taxon without a linked parent taxon.

### CODE REVIEW

File: spec/features/delete_taxon_spec.rb [hunk 1]
```diff
@@ -144,7 +143,6 @@ def then_the_taxon_is_deleted
 
   def then_the_taxon_is_restored
     expect(@put_content_request).to have_been_made
-    expect(@patch_links_request).to have_been_made
```
Comment (tijmenb) [inline spec/features/delete_taxon_spec.rb]: Why is this removed?

File: app/services/taxonomy/update_taxon.rb [hunk 2]
```diff
@@ -19,10 +19,13 @@ def publish(validate: true)
       end
 
       Services.publishing_api.put_content(content_id, payload)
-      Services.publishing_api.patch_links(
-        content_id,
-        links: { parent_taxons: Array(parent) }
-      )
+
+      if parent.present?
```
Comment (tijmenb) [inline app/services/taxonomy/update_taxon.rb]: Will this still allow the user to _remove_ the parent?

Comment (whoojemaflip) [inline app/services/taxonomy/update_taxon.rb]: This class handles the user's form submission, so if there is no parent taxon submitted in the UI then `taxon.parent` will be an empty string.

File: spec/features/delete_taxon_spec.rb (hunk 1 above)
Comment (whoojemaflip) [inline spec/features/delete_taxon_spec.rb]: lazy programmer

File: app/services/taxonomy/update_taxon.rb (hunk 2 above)
Comment (whoojemaflip) [inline app/services/taxonomy/update_taxon.rb]: ... so: yes

File: spec/features/delete_taxon_spec.rb (hunk 1 above)
Comment (whoojemaflip) [inline spec/features/delete_taxon_spec.rb]: resolved

File: app/services/taxonomy/update_taxon.rb (hunk 2 above)
Comment (whoojemaflip) [inline app/services/taxonomy/update_taxon.rb]: OIC
